api/server.py

```diff
@@ -61,22 +61,59 @@ async def websocket_connection(websocket: WebSocket, game_uuid: str,
                 data = await websocket.receive_json()
                 print("wow got message")
                 print(data)
+
                 # ignore message without 'type' key
                 # all messages must have a type
                 if "type" in data:
+

```

Comment:
I think if you wanted you could have just a single check here rather than one in each of move/suggestion/accusation

```
if type in types_to_validate: 
    validate 
    handle error
```
but up to you. Otherwise looks good